tests/foreman/ui/test_host.py

```diff
@@ -1701,3 +1704,190 @@ def test_positive_delete_libvirt(
+def class_gce_resource_with_image(
+        class_gce_template,
+        download_cert,
+        default_architecture,
+        module_os,
+        module_loc,
+        module_org
+):
+    with Session('gce_tests') as session:
+        # Until the CLI and API support is added for GCE,
+        # creating GCE CR from UI
+        cr_name = gen_string('alpha')
+        vm_user = gen_string('alpha')
+        session.computeresource.create({
+            'name': cr_name,
+            'provider': FOREMAN_PROVIDERS['google'],
+            'provider_content.google_project_id': settings.gce.project_id,
+            'provider_content.client_email': settings.gce.client_email,
+            'provider_content.certificate_path': settings.gce.cert_path,
+            'provider_content.zone.value': settings.gce.zone,
+            'organizations.resources.assigned': [module_org.name],
+            'locations.resources.assigned': [module_loc.name],
+        })
+    gce_cr = entities.AbstractComputeResource().search(
+        query={'search': 'name={}'.format(cr_name)})[0]
+    entities.Image(
+        architecture=default_architecture, compute_resource=gce_cr, name='autogce_img',
+        operatingsystem=module_os, username=vm_user, uuid=class_gce_template
+    ).create()
+    return gce_cr
+
+
+@pytest.fixture
+def class_gce_hostgroup(
+        module_org,
+        module_loc,
+        default_partition_table,
+        default_architecture,
+        module_os,
+        module_environment,
+        module_proxy,
+        class_gce_domain,
+        class_gce_resource_with_image,
+        module_lce,
+        module_content_view,
+):
+    return entities.HostGroup(
+        architecture=default_architecture,
+        compute_resource=class_gce_resource_with_image,
+        domain=class_gce_domain,
+        lifecycle_environment=module_lce,
+        content_view=module_content_view,
+        location=[module_loc],
+        environment=module_environment,
+        puppet_proxy=module_proxy,
+        puppet_ca_proxy=module_proxy,
+        content_source=module_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+        ptable=default_partition_table,
+    ).create()
+
+
+@tier4
+@skip_if_not_set('gce')
+def test_positive_gce_provision_end_to_end(
+        session,
+        module_org,
+        module_loc,
+        module_os,
+        class_gce_domain,
+        class_gce_hostgroup,
+        gce_client
+):
+    """Provision Host on GCE compute resource
+
+    :id: 8d1877bb-fbc2-4969-a13e-e95e4df4f4cd
+
+    :expectedresults: Host is provisioned successfully
+
+    :CaseLevel: System
+    """
+    name = gen_string('alpha').lower()
+    hostname = u'{0}.{1}'.format(name, class_gce_domain.name)
+    root_pwd = gen_string('alpha', 15)
+    with Session('gce_tests') as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision GCE Host
+        try:
```

Comment:
Just a quick suggestion to use helper :https://github.com/SatelliteQE/robottelo/pull/7261 in order to save the `yum update` time on client ... 

I'll review this PR bit later :)

Reply:
YUP, will do ! By the way the comments are given on that PR for some improvements.

Reply:
They are addressed , please check.